test/e2e/certman_operator_tests.go

```diff
@@ -62,4 +80,186 @@ var _ = Describe("Certman Operator", Ordered, func() {
 			return apiserver.Spec.ServingCerts.NamedCertificates[0].ServingCertificate.Name == secretName
 		}, pollingDuration, 30*time.Second).Should(BeTrue(), "Certificate secret should be applied to apiserver object")
 	})
+
+	It("remove finalizer from the ClusterDeployment", func(ctx context.Context) {
```

Comment:
It will trigger the finalizer deletion on a certain finalizer when the clusterdeployment is deleted.
I cannot quite understand the test here. It seems we try to set the finalizer to empty in the test itself? How would it related to the operator functionality?

Reply:
I've updated the test to verify the operator automatically ensures the finalizer is present when ClusterDeployment is not being deleted, rather than manually removing it. This now properly tests the controller's reconciliation behavior.